### PR TITLE
San 2635 use swarm events

### PR DESCRIPTION
**Pull request by anandkumarpatel**
- use swarm events
- add engine events
- make listener a unit test
- move test to unit folder
# review
- [x] @podviaznikov 
- [x] @bkendall 
# deps
- [x] ansible change
- [x] https://github.com/CodeNow/dock-init/pull/29
- [x] all docks have new hostname
- [x] https://github.com/CodeNow/dock-init/pull/30
# test
- [x]  ensure we get events to rabbit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,11 +58,11 @@
     "callback-count": "^0.2.0",
     "code": "^1.5.0",
     "docker-mock": "0.4.0",
-    "jshint": "2.5.10",
+    "jshint": "^2.9.1",
```
Comment (bkendall) [inline package.json]: `jshint` can probably be removed in favor of `standard`

File: lib/docker.js [hunk 2]
```diff
@@ -11,10 +11,8 @@ var join = require('path').join
 var Docker = require('dockerode')
 
 var dockerOpts = {
-  protocol: process.env.DOCKER_REMOTE_API_PROTOCOL,
-  host: process.env.DOCKER_REMOTE_API_HOST,
-  port: process.env.DOCKER_REMOTE_API_PORT,
-  timeout: parseInt(process.env.DL_DOCKER_TIMEOUT, 10)
```
Comment (podviaznikov) [inline lib/docker.js:17]: why do we delete timeout?

File: lib/workers/docker.event.publish.js [hunk 3]
```diff
@@ -109,37 +103,29 @@ DockerEventPublish._isContainerEvent = function (dockerEvent) {
 }
 
 /**
+ * should return true if the dockerEvent is a engine dockerEvent
+ */
+DockerEventPublish._isEngineEvent = function (dockerEvent) {
```
Comment (podviaznikov) [inline lib/workers/docker.event.publish.js]: where do you call this function?

File: lib/docker.js (hunk 2 above)
Comment (anandkumarpatel) [inline lib/docker.js:17]: there should be no timeout now, we want all logs from swarm no matter how long it takes to connect

File: lib/workers/docker.event.publish.js (hunk 3 above)
Comment (anandkumarpatel) [inline lib/workers/docker.event.publish.js]: never, I had it used before, now I dont, will remove

File: lib/docker.js (hunk 2 above)
Comment (podviaznikov) [inline lib/docker.js:17]: sounds good

**Comment by podviaznikov** [comment]
This looks great to me. Can you please after deploying it to prod delete all queues and close this task: https://runnable.atlassian.net/browse/SAN-3464?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+language: node_js
+node_js:
+- '4.2'
```
Comment (bkendall) [inline .travis.yml:3]: fwiw: `'4'` is sufficient since that's what the lts actually is

**Comment by bkendall** [comment]
I don't have major gripes about this... good luck!

**Pull request merged**